### PR TITLE
Removes creation of StatsD (and consequently EEC) containers in ActiveGate pods

### PR DESCRIPTION
**Pull request by StefanHauth**
# Description
Removes the creation of StatsD and EEC pods and all related config options from the operator code and hence from E2E tests.

## How can this be tested?
Build and deploy the operator. Deploy a DynaKube with AG capability `statsd-ingest` activated should give you an error message.

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/api/v1beta1/feature_flags_resources.go [hunk 1]
```diff
@@ -5,61 +5,6 @@ import (
...
-		log.Info("Problem parsing resource requirements for", "flagName", flagName, "val", val, "err", err.Error())
-		return nil
-	}
-
-	return &quantity
-}
-
 // +kubebuilder:object:generate=false
 type ResourceRequirementer interface {
 	Requests(corev1.ResourceName) *resource.Quantity
```
Comment (0sewa0) [inline src/api/v1beta1/feature_flags_resources.go:10]: none of these are needed, they are only used in the tests file for this file :)

**Review by 0sewa0 — CHANGES_REQUESTED** [review]
found `StatsdUrlSecretField    = "DT_STATSD_INGEST_URL"`
in `src/ingestendpoint/secret.go`

File: src/api/v1beta1/feature_flags_resources.go (hunk 1 above)
Comment (0sewa0) [reply src/api/v1beta1/feature_flags_resources.go:10]: so the whole `src/api/v1beta1/feature_flags_resources.go` can be removed (with its test)

File: src/api/v1beta1/feature_flags.go [hunk 2]
```diff
@@ -49,10 +49,8 @@ const (
 
 	// statsD
 
-	AnnotationFeatureUseActiveGateImageForStatsd = AnnotationFeaturePrefix + "use-activegate-image-for-statsd"
-	AnnotationFeatureCustomEecImage              = AnnotationFeaturePrefix + "custom-eec-image"
-	AnnotationFeatureCustomStatsdImage           = AnnotationFeaturePrefix + "custom-statsd-image"
-	AnnotationFeatureCustomSyntheticImage        = AnnotationFeaturePrefix + "custom-synthetic-image"
+	//TODO: is this statsd related as well?
```
Comment (chrismuellner) [inline src/api/v1beta1/feature_flags.go:52]: I don't see any more statsd feature flags

File: src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/config.go [hunk 3]
```diff
@@ -24,19 +23,13 @@ type initContainerModifier interface {
 	getInitContainers() []corev1.Container
 }
 
-var (
-	log = logger.Factory.GetLogger("activegate-statefulset-builder")
```
Comment (chrismuellner) [inline src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/config.go:28]: was this not used?

File: src/controllers/dynakube/version/version.go [hunk 4]
```diff
@@ -80,19 +76,7 @@ func (r *Reconciler) updateImages(ctx context.Context, spec updateSpec) error {
...
-		}
-	}
-
-	if spec.statsdUpdate {
-		err := imageUpdater.update(r.Dynakube.StatsdImage(), &r.Dynakube.Status.Statsd.VersionStatus, true)
-		if err != nil {
-			log.Error(err, "Failed to update StatsD image version")
-		}
-	}
+	// TODO: should there be a synthetic update
```
Comment (chrismuellner) [inline src/controllers/dynakube/version/version.go:79]: remove these synthetic todos, this will need to be done when there is full support for synthetic

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/config.go (hunk 3 above)
Comment (StefanHauth) [reply src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/config.go:28]: Only by `ExtensionControllerModifier`, which got thrown out.

File: src/controllers/dynakube/version/version.go (hunk 4 above)
Comment (0sewa0) [inline src/controllers/dynakube/version/version.go:79]: I still see this todo 🙃

**Review by 0sewa0 — APPROVED** [review]

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**